libs/vm/include/vm/object.hpp

```diff
@@ -42,42 +42,44 @@ class Address;
 struct String;
 
 template <typename T>
-using IsPrimitive =
-    type_util::IsAnyOf<T, void, bool, int8_t, uint8_t, int16_t, uint16_t, int32_t, uint32_t,
-                       int64_t, uint64_t, float, double, fixed_point::fp32_t, fixed_point::fp64_t>;
+static constexpr bool IsPrimitive =
+    type_util::IsAnyOfV<T, void, bool, int8_t, uint8_t, int16_t, uint16_t, int32_t, uint32_t,
+                        int64_t, uint64_t, float, double, fixed_point::fp32_t, fixed_point::fp64_t>;
 
 template <typename T, typename R = void>
-using IfIsPrimitive = std::enable_if_t<IsPrimitive<std::decay_t<T>>::value, R>;
+using IfIsPrimitive = std::enable_if_t<IsPrimitive<std::decay_t<T>>, R>;
 
 template <typename T>
-using IsObject = std::is_base_of<Object, T>;
+static constexpr bool IsObject = std::is_base_of<Object, T>::value;
 
 template <typename T>
-struct IsPtr : std::false_type
+struct IsPtrImpl : std::false_type
 {
 };
 template <typename T>
-struct IsPtr<Ptr<T>> : std::true_type
+struct IsPtrImpl<Ptr<T>> : std::true_type
 {
 };
 
+template <typename T>
+static constexpr bool IsPtr = IsPtrImpl<T>::value;
+
 template <typename T, typename R = void>
-using IfIsPtr = std::enable_if_t<IsPtr<std::decay_t<T>>::value, R>;
+using IfIsPtr = std::enable_if_t<IsPtr<std::decay_t<T>>, R>;
 
 template <typename T>
-using IsVariant = std::is_base_of<Variant, T>;
+static constexpr bool IsVariant = std::is_base_of<Variant, T>::value;
 
 template <typename T>
-using IsAddress = std::is_base_of<Address, T>;
+static constexpr bool IsAddress = std::is_base_of<Address, T>::value;
 
 template <typename T>
-using IsString = std::is_base_of<String, std::decay_t<T>>;
+static constexpr bool IsString = std::is_base_of<String, std::decay_t<T>>::value;
 
 template <typename T, typename R = void>
 using IfIsExternal = std::enable_if_t<
-    (!IsPtr<std::decay_t<T>>::value) && (!IsObject<std::decay_t<T>>::value) &&
-        (!IsVariant<std::decay_t<T>>::value) && (!IsPrimitive<std::decay_t<T>>::value) &&
-        (!IsString<std::decay_t<T>>::value) && (!IsAddress<std::decay_t<T>>::value),
+    (!IsPtr<std::decay_t<T>>)&&(!IsObject<std::decay_t<T>>)&&(!IsVariant<std::decay_t<T>>)&&(
```

Comment:
Restore the previous spacing style, it was easier to read.

Reply:
The formatting is fully under control of our styling script (`apply_style.py`) which is executed automatically as the part of our CI. Any deviation from that styling will result in failure in CI.

Reply:
While true, you can always put in the exception guards if you feel like it is really making a bad job of the formatting

Reply:
I can change this using the guards.
But, to be honest, code formatting/style is quite personal preference endeavour and can vary from person to person.

However, if these kind of comments are going to be a thing in long run, it is a bit contra productive to do this manually  (either after `apply_style.py` run, or during PR review where making&handling them is quite expensive). We should rather make a change to `apply_style.py` which shall be point of reference for code style.